### PR TITLE
feat: replace form components

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by Sboonny** [comment]
This can't be merged without FormGroup, because it doesn't get the validationstate from the parent.

image

**Comment by ahmaxed** [comment]
Solid work here. This is what stood out:
1- the color of the help-block under the email field does not match the danger-color.
Screenshot 2023-09-20 at 3 15 12 PM

2- on the internet component, some validation are getting the tick some don't. Also the tick is outside of the form field. 
Screenshot 2023-09-20 at 3 20 39 PM

This is unrelated to this pr, but we should display all validations in the same way across the platform. Either tick for all valid inputs or for none, the input feedback should either be in a help block or in an alert, etc.

File: client/src/client-only-routes/show-user.tsx [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
-} from '@freecodecamp/react-bootstrap';
+import { Panel, Button, Col, Row } from '@freecodecamp/react-bootstrap';
 import React, { useState } from 'react';
 import Helmet from 'react-helmet';
 import type { TFunction } from 'i18next';
 import { Trans, withTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
-
+import { FormGroup, FormControl, ControlLabel } from '@freecodecamp/ui';
```
Comment (Sboonny) [inline client/src/client-only-routes/show-user.tsx:8]:
```suggestion
import { FormGroup, FormControl, ControlLabel } from '@freecodecamp/ui';

```

**Review by ahmaxed — APPROVED** [review]
LGMT,
My only nitpick is that text area should get a hight of auto so it looks different from an input.
Screenshot 2023-10-11 at 2 45 43 PM
Screenshot 2023-10-11 at 2 45 51 PM

**Comment by Sboonny** [comment]
> My only nitpick is that text area should get a hight of auto so it looks different from an input.

I forgot to mention it here. I am aware of the specificity issue that stop the textarea from getting `h-8` className, and plan to fix it with the decoupling of the component, when I turn `FormControl` to `Input` and `Textarea`.

**Comment by ahmaxed** [comment]
I think the textarea is getting the h-8, but is should not.

File: tools/ui-components/src/form-control/form-control.tsx [hunk 2]
```diff
@@ -9,47 +9,24 @@ import { FormControlProps } from './types';
...
-      placeholder={placeholder}
-      type={type}
-      {...restProps}
-    />
-  );
+  const classes = [className, defaultClasses, variantClass].join(' ');
+
+  return <Component id={id || controlId} className={classes} {...props} />;
 };
+<FormControl />;
```
[the remark this answers is not in the document]
Comment (Sboonny) [reply tools/ui-components/src/form-control/form-control.tsx:29]: yeah, I was using it to test the types, before building the component library for the client. Properly added it through `git add -A` 👍

**Review by huyenltnguyen — COMMENTED** [review]
Just a couple more 😄

**Review by huyenltnguyen — APPROVED** [review]
LGTM.

I think we will need to wait for https://github.com/freeCodeCamp/freeCodeCamp/pull/51980 as the CI failures appear to be caused by the Node upgrade 😓

**Pull request merged**